joomla/joomla-cms
# [4.0] Cleanup and refactor JLIB_DATABASE_ERROR_... strings

The `lib_joomla.ini` contains a lot of JLIB_DATABASE_ERROR strings which are nowhere used in code. That is just unneeded translator work.
There are also some strings where the class was injected, but the method was part of the string. The method itself shouldn't be translated. This PR adds a new generic string where the method and error message is injected.
In the nested table class, there is a function which requires an error message, but never uses it. I've changed the method so the argument is optional and removed the error messages. The strings used wouldn't work as error message anyway since they expect a parameter.

### Summary of Changes
* The unused strings are removed from the INI files.
* `JLIB_DATABASE_ERROR_MOVE_FAILED` and `JLIB_DATABASE_ERROR_DELETE_FAILED` and  are changed to the new generic `JLIB_DATABASE_ERROR_METHOD_FAILED`
* The function `_runQuery` in libraries/src/Table/Nested.php was changed to have an optional error message instead of required. The messages strings are removed.
* As a bonus, the language class now uses the correct (existing) string for when the SEF code is already present.

### Testing Instructions
As with the other PRs, it's hard to actually test this. Code review should do.
You can try to see if the strings are indeed not used in the CMS.

### Documentation Changes Required
None
